builder_design_pattern/src/Main.java

```diff
@@ -0,0 +1,12 @@
+public class Main {
+    public static void main(String[] args) {
+        Director director = new Director();
+        PaySlip.Builder builder = new PaySlip.Builder();
+        director.createAssociateSoftwareEngineerPaySlip(builder, "Akash", "7894456213", "ERTF4563F");
+        PaySlip paySlip1 = builder.build();
```

Comment:
This works but it is very difficult to read/review. The builder is passed to `director.createAssociateSoftwareEngineerPaySlip()` which only updates the builder with various values.

We could have just returned `paySlip1` from `directory.createAssociateSoftwareEngineerPaySlip()`. Makes code much more readable.